src/PatternkitDrupalTwigLib.php

```diff
@@ -195,14 +247,123 @@ protected function getRawMetadata() {
    * @return string
    *   The rendered pattern HTML.
    */
-  public function getRenderedPatternMarkup(PatternkitPattern $pattern,
-    PatternkitEditorConfig $config) {
+  public function getRenderedPatternMarkup(
+    PatternkitPattern $pattern,
+    PatternkitEditorConfig $config
+  ) {
     if (empty($pattern->filename) || empty($config->fields)) {
       return '';
     }
     $template = $pattern->filename;
     $variables = $config->fields;
-    return twig_render_template($template, $variables);
+
+    return $this->renderTwigTemplate($template, $variables);
+  }
+
+  /**
+   * Returns a singleton version of the twig template engine.
+   *
+   * @return Twig_Environment
+   *   Twig environment object.
+   *
+   * @throws \Twig_Error_Loader
+   *   Twig engine instance object.
+   */
+  public function getTwigInstance() {
+    static $twig_engine;
```

Comment:
Should this be determinate on if `tfd7` is already loaded or not? I'm wondering about the overhead of potentially bootstrapping the twig engine twice, although we probably need to test if it actually makes an impact or not.

Reply:
I replaced tfd7 entirely, but now that I think of it, you guys probably use it for other stuff. TFD7 does a /lot/ of extra work here. 

Invoking twig with the namespacing requirements means you're paying for the class invocation regardless of approach, so at the very least this work is siloed from that code.